src/veh_interact.cpp

```diff
@@ -316,30 +315,19 @@ task_reason veh_interact::cant_do (char mode)
         } );
         has_tools = true; // checked later
         break;
-    case 'f': // refill mode
-        if (!ptanks.empty()) {
-            std::vector<vehicle_part *>::iterator iter;
-            for (iter = ptanks.begin(); iter != ptanks.end(); ) {
-                if ((*iter)->hp > 0 &&
-                    g->refill_vehicle_part(*veh, *iter, true)) {
-                    pass_checks = true;
-                    iter++;
-                } else {
-                    iter = ptanks.erase(iter);
-                }
-            }
-            if(pass_checks) {
-                return CAN_DO;
-            }
-        }
 
-        // if refillable parts exist but can't be refilled for some reason.
-        if (has_ptank) {
-            return CANT_REFILL;
-        }
+    case 'f': {
+        // first find all possible ammo
+        auto opts = g->u.items_with( []( const item &e ) { return e.is_ammo(); } );
```

Comment:
> We don't actually have bleach tanks however or indeed tanks that hold anything other than ammo. This and any future implementation is greatly simplified by following this rule as opposed to supporting arbitrary functionality that we currently don't use. YAGNI.

YAGNI is my argument as well. What part of the ammo data is required by the tank vehicle part? Container items don't require their content to be ammo. Tank vehicle parts should behave similar.

> If we we did want bleach tanks we should support them the same way we do water tanks, by providing an  ammo  field in the JSON definition.

Now you're advocating a hack. Again, why does a liquid need ammo data just to store it in a vehicle tank?